### PR TITLE
feat(expressive-list-container): adds expressive list container component

### LINKED ISSUE #1083: Enhancement[ expressive list]: add expressive list to component library
The "expressive list" pattern is used a lot throughout gatsby sites. There are variations in how the rule is designed, spacing and behavior on tablet, so a component with predefined specs would help consistency. 

examples of the list in use:
- in full width, top nav, pages:
https://www.ibm.com/design/practices/
https://www.ibm.com/design/teams/
https://www.ibm.com/design/impact/

- in pages with a left nav:
https://w3.ibm.com/design/about/overview/
https://w3.ibm.com/design/leadership/overview/
https://w3.ibm.com/design/community/
https://www.ibm.com/brand/architecture
https://www.ibm.com/brand/systems/services/strategy
https://www.ibm.com/brand/systems/it-infrastructure/expression/strategy/
https://www.ibm.com/design/language/philosophy/principles/
https://www.ibm.com/design/language/layout/overview/
https://www.ibm.com/design/language/illustration/overview/ (with a white background)
https://www.ibm.com/able/toolkit/plan (with a white background and a pictogram)

specs excerpts:

Screenshot 2021-01-05 at 11 56 45
Screenshot 2021-01-05 at 11 57 01
Screenshot 2021-01-05 at 11 57 14

sketch file:
[Expressive list specs_ds3.zip](https://github.com/carbon-design-system/gatsby-theme-carbon/files/5769401/Expressive.list.specs_ds3.zip)

### PR DESCRIPTION
**Pull request by aledavila**
Closes #1083

This creates the ExpressiveListContainer which is seperate from the ExpressiveList due to differing uses and styles. This component allows a user to insert whatever they want into the component. 

Screen Shot 2021-12-13 at 2 53 33 PM

#### Changelog

**New**

- Added `ExpressiveListContainer`

### CODE REVIEW

**Comment by DianaStanciulescu** [comment]
making this a separate component makes perfect sense. the only tricky thing here is forcing the space between the rule and whatever comes after to stay at 24px, even if the actual component may have a larger or smaller top spacer.

**Comment by alisonjoseph** [comment]
I think I'm still confused on why these are separate components, but maybe I'm missing something. 😅 

The spacing differences will need to be accounted for regardless, there is nothing stopping someone from just placing text in this component, or adding a child component to the other component. I know @aledavila is out this week, and then I'm out next week, so we won't be able to sync up on this until the new year. 

Maybe the three of us can jump on a quick call after the holiday to talk through it. 😄

**Comment by DianaStanciulescu** [comment]
hi! based on our conversation:
- try to see if the top margin for the first child can be 24px, when the child is text, card, image, etc. (maybe try minus 24)
- the top margin for the rule is 48px
- the titles should left align to the rest of the text on the page - so 16px to the right

**Comment by alisonjoseph** [comment]
What is the blocker for v11? nested grid support?

**Comment by aledavila** [comment]
@alisonjoseph yes that is true I guess we can merge it since the container does work. I'll make a note of its use

**Comment by aledavila** [comment]
@alisonjoseph Yea I’ll put the disclaimer for this real quick and it’s fine to merge. Do you think something along the lines of saying this doesn’t work with component that are built using the grid and requires additional custom styles for it ?

**Comment by alisonjoseph** [comment]
Something around nested grid support will be supported in v11, currently you will need to add custom styles to nest grids?

**Comment by alisonjoseph** [comment]
@aledavila status of this? Can we add that message and get this merged?

**Comment by aledavila** [comment]
@alisonjoseph yes ready for review. Please check if that message makes sense.

File: packages/example/src/pages/components/ExpressiveListContainer.mdx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+import { Connect } from '@carbon/pictograms-react';
+
+<PageDescription>
+
+The `<ExpressiveListContainer>` component can be used as a customizable component. 
+  
+***Disclaimer: Nested grid support will be supported in v11. Currently this component requires custom styles to nest most of the gatsby components into the container.***
+
+</PageDescription>
```
Comment (alisonjoseph) [inline packages/example/src/pages/components/ExpressiveListContainer.mdx:14]:
```suggestion
<PageDescription>

The `<ExpressiveListContainer>` component can be used as a customizable component. 

</PageDescription>

<InlineNotification kind="warning">

**Disclaimer:** Nested grid support will be supported in v11. Currently this component requires custom styles to nest most of the gatsby components into the container.

</InlineNotification>
```

**Review by alisonjoseph — APPROVED** [review]
Message is perfect, I just updated it to use the inline notification instead. 😄

**Comment by aledavila** [comment]
@alisonjoseph oh that was a good idea. I'll cut a version release for all the updates this afternoon.

**Pull request merged**